airflow/providers/amazon/aws/operators/sagemaker_processing.py

```diff
@@ -0,0 +1,121 @@
+# under the License.
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.operators.sagemaker_base import SageMakerBaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class SageMakerProcessingOperator(SageMakerBaseOperator):
+    """
+    Initiate a SageMaker processing job.
+
+    This operator returns The ARN of the processing job created in Amazon SageMaker.
+
+    :param config: The configuration necessary to start a processing job (templated).
+
+        For details of the configuration parameter see :py:meth:`SageMaker.Client.create_processing_job`
+    :type config: dict
+    :param aws_conn_id: The AWS connection ID to use.
+    :type aws_conn_id: str
+    :param wait_for_completion: If wait is set to True, the time interval, in seconds,
+        that the operation waits to check the status of the processing job.
+    :type wait_for_completion: bool
+    :param print_log: if the operator should print the cloudwatch log during processing
+    :type print_log: bool
+    :param check_interval: if wait is set to be true, this is the time interval
+        in seconds which the operator will check the status of the processing job
+    :type check_interval: int
+    :param max_ingestion_time: If wait is set to True, the operation fails if the processing job
+        doesn't finish within max_ingestion_time seconds. If you set this parameter to None,
+        the operation does not timeout.
+    :type max_ingestion_time: int
+    :param action_if_job_exists: Behaviour if the job name already exists. Possible options are "increment"
+        (default) and "fail".
+    :type action_if_job_exists: str
+    """
+
+    integer_fields = [
+        ['ProcessingResources', 'ClusterConfig', 'InstanceCount'],
+        ['ProcessingResources', 'ClusterConfig', 'VolumeSizeInGB'],
+        ['StoppingCondition', 'MaxRuntimeInSeconds']
+    ]
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 wait_for_completion=True,
+                 print_log=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 action_if_job_exists: str = "increment",  # TODO use typing.Literal for this in Python 3.8
+                 *args, **kwargs):
+        super().__init__(config=config, *args, **kwargs)
+
+        self.wait_for_completion = wait_for_completion
+        self.print_log = print_log
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+
+        if action_if_job_exists in ("increment", "fail"):
+            self.action_if_job_exists = action_if_job_exists
+        else:
+            raise AirflowException(
+                "Argument action_if_job_exists accepts only 'increment' and 'fail'. "
+                f"Provided value: '{action_if_job_exists}'."
+            )
```

Comment:
```suggestion
        if action_if_job_exists not in ("increment", "fail"):
            raise AirflowException(
                "Argument action_if_job_exists accepts only 'increment' and 'fail'. "
                f"Provided value: '{action_if_job_exists}'."
            )
        self.action_if_job_exists = action_if_job_exists
```
And you want to put the validation at the very top, because it is essential. Without a proper value this method don't need to go on.

Reply:
Fixed.